stdlib/public/SwiftShims/FloatingPointShims.h

```diff
@@ -0,0 +1,123 @@
+//===--- FloatingPointShims.h ---------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+//  Using the Darwin (or Glibc) module in the core stdlib would create a
+//  circular dependency, so instead we import these declarations as part of
+//  SwiftShims.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_STDLIB_SHIMS_FLOATINGPOINTSHIMS_H
+#define SWIFT_STDLIB_SHIMS_FLOATINGPOINTSHIMS_H
+
+#include "SwiftStdint.h"
+#include "SwiftStddef.h"
+#include "Visibility.h"
+
+#if __has_builtin(__builtin_remainderf)
+#define SWIFT_HAVE_BUILTIN_REMAINDERF 1
+#else
+#define SWIFT_HAVE_BUILTIN_REMAINDERF 0
+#endif
+
+#if __has_builtin(__builtin_sqrtf)
+#define SWIFT_HAVE_BUILTIN_SQRTF 1
+#else
+#define SWIFT_HAVE_BUILTIN_SQRTF 0
+#endif
+
+#if __has_builtin(__builtin_remainder)
+#define SWIFT_HAVE_BUILTIN_REMAINDER 1
+#else
+#define SWIFT_HAVE_BUILTIN_REMAINDER 0
+#endif
+
+#if __has_builtin(__builtin_sqrt)
+#define SWIFT_HAVE_BUILTIN_SQRT 1
+#else
+#define SWIFT_HAVE_BUILTIN_SQRT 0
+#endif
+
+#if !SWIFT_HAVE_BUILTIN_REMAINDERF || !SWIFT_HAVE_BUILTIN_SQRTF ||             \
+    !SWIFT_HAVE_BUILTIN_REMAINDER || !SWIFT_HAVE_BUILTIN_SQRT
+#include <math.h>
+#endif
+
+#if __has_feature(nullability)
+#pragma clang assume_nonnull begin
+#endif
+
+#ifdef __cplusplus
+namespace swift {
+extern "C" {
+#endif
+
+// Math library functions
+static inline SWIFT_ALWAYS_INLINE float _swift_stdlib_remainderf(float _self,
+                                                                 float _other) {
+#if SWIFT_HAVE_BUILTIN_REMAINDERF
+  return __builtin_remainderf(_self, _other);
+#else
+  return remainderf(_self, _other);
+#endif
+}
+
+static inline SWIFT_ALWAYS_INLINE float _swift_stdlib_squareRootf(float _self) {
+#if SWIFT_HAVE_BUILTIN_SQRTF
+  return __builtin_sqrtf(_self);
+#else
+  return sqrtf(_self);
+#endif
+}
+
+static inline SWIFT_ALWAYS_INLINE double
+_swift_stdlib_remainder(double _self, double _other) {
+#if SWIFT_HAVE_BUILTIN_REMAINDER
+  return __builtin_remainder(_self, _other);
+#else
+  return remainder(_self, _other);
+#endif
+}
+
+static inline SWIFT_ALWAYS_INLINE double
+_swift_stdlib_squareRoot(double _self) {
+#if SWIFT_HAVE_BUILTIN_SQRT
+  return __builtin_sqrt(_self);
+#else
+  return sqrt(_self);
```

Comment:
Would be nice, I have no idea how though. Unfortunately 4.0 shipped with the mangling but so most people are gonna get crashers building swift on windows right now.
N.b this doesn't just affect the runtime but also other files (e.g. remoteast.cpp)